[DOC] Add section on automated contributions (#6101)
nilearn/nilearn: CONTRIBUTING.rst

Please see the :ref:`contributing_code` section for more detailed information, including
instructions for  `Setting up your environment`_ and a description of the `Contribution Guidelines`_.

Automated contributions policy
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Contributing to Nilearn requires human judgment, contextual understanding,

Reviewer: ```suggestion
Contributing to Nilearn requires human judgement, contextual understanding,
```

Reviewer 2: let's stick to the american spelling

Reviewer: Ah, that's my ignorence. I did not know that!

Reviewer 2: Me neither.I had to look it up.

and familiarity with Nilearn's structure and goals.

> It is not suitable for automatic processing by AI tools.

Reviewer 2: @hndgzkn 
you mentioned that you did not find it clear enough that this was about 100% agent driven PR

do you feel you can help clarify this?

> It is not suitable for automatic processing by AI tools.

Reviewer: ```suggestion
It is not suitable for automated AI coding agent driven contributions.
```

> Instead, we value authentic engagement in our community.

Reviewer: ```suggestion
Instead, we value authentic human engagement in our community.
```

- Do not post output from Large Language Models or similar generative AI
  as comments on our GitHub or our Discord server,
  as such comments tend to be formulaic and low content.

- If you use generative AI tools
  as an aid in developing code or documentation changes,

>   ensure that you fully understand and can explain the proposed changes.

Reviewer: ```suggestion
  ensure that you fully understand and can explain the proposed changes, and it complies with Nilearn's code structure and coding conventions.
```

Please do not submit issues or pull requests generated by fully-automated tools; i.e., without human review.
Maintainers reserve the right, at their sole discretion, to close such submissions and to block any account responsible for them.

> If you have used AI tools, please declare their use in your issue or pull request description.

Reviewer 3: I saw [this project's page](https://www.openms.org/news/gsoc2026/) recently where they use language that still remains supportive of AI tool usage:

> ### Make clear in any submission (code, communications, etc.) that you used AI tools.
> This can simply be a line in the PR description saying “Made with AI tools”. It should be clear that this isn’t meant to pass judgement on you, work made with AI tools has a very different set of typical failure modes than work done entirely by humans, it helps us to understand where mistakes may have come from.

Maybe we can borrow some of that if we think it's useful to explain why we require declaration of AI tool usage.

Reviewer 2: I think that can be combined with a change to the PR template asking if any AI was used to create the PR
